sdks/python/apache_beam/yaml/yaml_provider.py

```diff
@@ -665,6 +675,27 @@ def __init__(self, windowing):
     def expand(self, pcoll):
       return pcoll | self._window_transform
 
+    @staticmethod
+    def _parse_time_unit(value, name):
```

Comment:
While for a lot of the yaml construction, integration tests are more natural than unit tests, this function is of the right form to add some good unit tests too.

Reply:
Added some tests - required moving the classes around so they are visible outside the file.